upup/pkg/fi/cloudup/urls.go

```diff
@@ -17,74 +17,162 @@ limitations under the License.
 package cloudup
 
 import (
+	"net/url"
 	"os"
 	"strings"
 
+	"path"
+
 	"github.com/golang/glog"
 	"k8s.io/kops"
+	"k8s.io/kops/pkg/assets"
+	"k8s.io/kops/util/pkg/hashing"
 )
 
-// baseUrl caches the BaseUrl value
-var baseUrl string
+var kopsUrl = "https://kubeupv2.s3.amazonaws.com/kops/" + kops.Version + "/"
```

Comment:
This seems risky.  What are the ordering rules for evaluation here?

Reply:
> This seems risky. What are the ordering rules for evaluation here?

You mean adding the string?  I can do a fmt.sprintf

Reply:
I mean evaluating `kops.Version` in another global variable, when kops.Version is substituted.  What are the ordering rules for evaluation here?

Reply:
Ah, will move it to the func